transport/conn.go

```diff
@@ -307,34 +318,42 @@ func OpenConn(addr string, localAddr *net.TCPAddr, cfg ConnConfig) (*Conn, error
 	return WrapConn(tcpConn)
 }
 
+func (c *Conn) closeOnce() {
+	c.once.Do(c.Close)
+}
+
 // WrapConn transforms tcp connection to a working Scylla connection.
 // If error and connection are returned the connection is not valid and must be closed by the caller.
 func WrapConn(conn net.Conn) (*Conn, error) {
 	m := new(connMetrics)
-	c := &Conn{
+	var c Conn
+	c = Conn{
 		conn: conn,
 		w: connWriter{
 			conn:      bufio.NewWriterSize(conn, ioBufferSize),
 			requestCh: make(chan request, requestChanSize),
 			metrics:   m,
+			connClose: c.closeOnce,
```

Comment:
To make this assign possible I had to declare `c` earlier in code that's why I have added
`var c Conn`, is it worth it?
Other option is just keep it as it was and add
```
c.w.connClose = c.connClose
c.r.connClose = c.connClose
```
after the `&Conn` declaration, what do you think?